src/doc/Line.ts

```diff
@@ -11,23 +11,30 @@ export type LineIds = Map<string, Line>;
 export type LineInfo = {ranges: LineRanges, ids: LineIds};
 
 interface Line {
+  id: string;
   attributes: AttributeMap;
   content: Delta;
   length: number;
 }
 
 
 namespace Line {
-  export function iterator(lines: Line[]) {
-    return new Iterator(lines);
+  export function iterator(lines: Line[], lineIds?: LineIds) {
+    return new Iterator(lines, lineIds);
+  }
+
+  export function linesToLineIds(lines: Line[]) {
+    const lineIds = new Map();
+    lines.forEach(line => lineIds.set(line.id || Line.createId(lineIds), line));
+    return lineIds;
   }
 
   export function length(line: Line): number {
     return line.length;
   }
 
   export function getId(line: Line): string {
```

Comment:
There is an opportunity to remove this function entirely. Is that desirable?

Reply:
The logic is used in 2 places, so this dedupes it into one. I think it is worth keeping.

Reply:
Let's call it deprecated. I'm not sure if others have code that relies on it.